classes/Http/Controller/TalkController.php

```diff
@@ -15,415 +15,38 @@
 
 use OpenCFP\Application\NotAuthorizedException;
 use OpenCFP\Application\Speakers;
-use OpenCFP\Domain\CallForPapers;
-use OpenCFP\Domain\Model\Talk;
-use OpenCFP\Domain\Services\Authentication;
-use OpenCFP\Http\Form\TalkForm;
-use OpenCFP\Http\View\TalkHelper;
-use Swift_Message;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\Session;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
 use Twig_Environment;
 
 class TalkController extends BaseController
 {
-    /**
-     * @param $requestData
-     *
-     * @return TalkForm
-     */
-    private function getTalkForm($requestData): TalkForm
-    {
-        /** @var TalkHelper $helper */
-        $helper = $this->service(TalkHelper::class);
+    private $speakers;
 
-        $options = [
-            'categories' => $helper->getTalkCategories(),
-            'levels'     => $helper->getTalkLevels(),
-            'types'      => $helper->getTalkTypes(),
-        ];
-
-        /** @var \HTMLPurifier $htmlPurifier */
-        $htmlPurifier = $this->service('purifier');
+    public function __construct(Speakers $speakers, Twig_Environment $twig, UrlGeneratorInterface $urlGenerator)
+    {
+        $this->speakers = $speakers;
 
-        return new TalkForm($requestData, $htmlPurifier, $options);
+        parent::__construct($twig, $urlGenerator);
     }
 
     /**
      * Controller action for viewing a specific talk
      *
      * @param Request $request
      *
-     * @return mixed
+     * @return Response
      */
     public function viewAction(Request $request)
     {
-        /* @var Speakers $speakers */
-        $speakers = $this->service('application.speakers');
-
         try {
             $talkId = (int) $request->get('id');
-            $talk   = $speakers->getTalk($talkId);
+            $talk   = $this->speakers->getTalk($talkId);
         } catch (NotAuthorizedException $e) {
             return $this->redirectTo('dashboard');
         }
 
         return $this->render('talk/view.twig', \compact('talkId', 'talk'));
     }
-
-    public function editAction(Request $request)
```

Comment:
~~Not sure what is happening here, has this functionality moved elsewhere?~~

Ah, I see, the `TalkFormController` below!

@BackEndTea @chartjes @mdwheele 

What do you think about extracting *actions* sooner or later, as functors?

This would have the following advantage

* dependencies are limited to what is needed for a single action (right now we would inject dependencies required for all actions, even if not all of them are used for a single request)
* routing definitions are more transparent (as we can ideally just use the class name, no more hard-coded class names concatenated with `:` and the action)

Reply:
Yes, I've split the controller because we would inject too many unnecessary dependencies otherwise. And yes, I'd prefer single-action controllers as well, but that would've been out of scope for this PR.

Reply:
Absolutely, @derrabus! Happy to start dissecting! 👍

Reply:
I too think this would be a great idea once we get this PR merged.